src/utils/Processing.js

```diff
@@ -12,30 +13,85 @@ export function fillTank(rawFSHes, config) {
+      helperReturn.version = database.version;
+      if (existingObjectStores.length === 0 || dependencyArr.length === 0) {
+        helperReturn.shouldUpdate = true;
+        database.close();
+        resolve(helperReturn);
+      } else {
+        for (let i = 0; i < dependencyArr.length; i++) {
+          let dependency = dependencyArr[i][0];
+          let id = dependencyArr[i][1];
+          if (!existingObjectStores.contains(`${dependency}${id}`)) {
+            helperReturn.shouldUpdate = true;
+          }
+        }
+        database.close();
+        resolve(helperReturn);
+      }
+    };
+    OpenIDBRequest.onupgradeneeded = function (event) {
+      database = event.target.result;
+    };
+    OpenIDBRequest.onerror = function (event) {
+      reject(event);
+    };
+  });
+}
+
+export async function loadExternalDependencies(FHIRdefs, version, dependencyArr) {
+  return new Promise((resolve, reject) => {
+    let database = null;
+    let newDependencies = [];
     let finalDefs = FHIRDefinitions;
     const OpenIDBRequest = indexedDB.open('FSH Playground Dependencies', version);
     // If successful the database exists
     OpenIDBRequest.onsuccess = async function (event) {
       database = event.target.result;
-      const resources = [];
-      if (shouldUnzip) {
-        await unzipDependencies(resources);
-        await loadDependenciesInStorage(database, resources);
+      let findR4 = findIndex(dependencyArr, (elem) => elem[0] === 'hl7.fhir.r4.core' && elem[1] === '4.0.1');
+      if (findR4 < 0) {
+        dependencyArr.push(['hl7.fhir.r4.core', '4.0.1']);
       }
-      finalDefs = await loadAsFHIRDefs(FHIRdefs, database);
+      for (let i = 0; i < dependencyArr.length; i++) {
+        let resources = [];
+        let shouldUnzip = false;
+        let dependency = dependencyArr[i][0];
+        let id = dependencyArr[i][1];
+        if (newDependencies.includes(`${dependency}${id}`)) {
+          shouldUnzip = true;
+        }
+        if (shouldUnzip) {
+          resources = await unzipDependencies(resources, dependency, id);
+          await loadDependenciesInStorage(database, resources, dependency, id);
+        }
+        finalDefs = await loadAsFHIRDefs(FHIRdefs, database, dependency, id);
+      }
+      database.close();
       resolve(finalDefs);
     };
     // If upgrade is needed to the version, the database does not yet exist
     OpenIDBRequest.onupgradeneeded = function (event) {
-      shouldUnzip = true;
+      dependencyArr.push(['hl7.fhir.r4.core', '4.0.1']);
       database = event.target.result;
-      database.createObjectStore('resources', {
-        keyPath: ['id', 'resourceType']
-      });
+      let existingObjectStores = database.objectStoreNames;
+      for (let i = 0; i < dependencyArr.length; i++) {
+        let dependency = dependencyArr[i][0];
+        let id = dependencyArr[i][1];
+        if (!existingObjectStores.contains(`${dependency}${id}`)) {
```

Comment:
Just for record keeping, Sam and I talked about this one offline, and we decided it didn't really save us much in terms of run time, and the only repetition is with the if check for existing object store names. Since that check seemed simple enough, we figured we'd keep this as it is now.